nativelib/src/main/scala/scala/scalanative/unsafe/package.scala

```diff
@@ -196,30 +196,54 @@ package object unsafe extends unsafe.UnsafePackageCompat {
     }
   }
 
+  private def toCStringNonNull(cb: CharBuffer, cs: Charset, charSize: CInt)(
+      implicit z: Zone
+  ): Ptr[Byte] = {
+    val buf = cs.encode(cb)
+    val off = buf.position()
+
+    val size = (buf.limit() - off).toUSize
+    val cstr = z.alloc(size + charSize.toUSize)
```

Comment:
I suggest adding a one line comment above the z.alloc
stating that z.alloc always zeros memory returned.
The extra bytes allocated become/are the C `\0` NUL terminator.  

This relies on the plausable assumption the the memcpy
line below is correct.  If it is not, we have bigger problems.